apps/judicial-system/message-handler/src/app/prosecutorDocumentsDelivery.service.ts

```diff
@@ -0,0 +1,63 @@
+import fetch from 'node-fetch'
+
+import { Inject, Injectable } from '@nestjs/common'
+
+import type { ConfigType } from '@island.is/nest/config'
+import { logger } from '@island.is/logging'
+
+import { appModuleConfig } from './app.config'
+
+@Injectable()
+export class ProsecutorDocumentsDeliveryService {
+  constructor(
+    @Inject(appModuleConfig.KEY)
+    private readonly config: ConfigType<typeof appModuleConfig>,
+  ) {}
+
+  async deliverProsecutorDocuments(caseId: string): Promise<void> {
+    logger.debug(`Delivering prosecutor documents for case ${caseId} to court`)
+
+    return fetch(
+      `${this.config.backendUrl}/api/internal/case/${caseId}/deliverProsecutorDocuments`,
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          authorization: `Bearer ${this.config.backendAccessToken}`,
+        },
+      },
+    )
+      .then(async (res) => {
+        const response = await res.json()
+
+        if (res.ok) {
+          logger.debug(
+            `Delivered prosecutor documents for case ${caseId} to court`,
+          )
+
+          if (!response.requestDeliveredToCourt) {
+            logger.error(
+              `Failed to deliver the request for case ${caseId} to court`,
+            )
+          }
+
+          return
+        }
+
+        logger.error(
```

Comment:
I think it's a bit weird to have this code here and in the .catch. Consider throwing an error here instead.

Reply:
I can understand where you are coming from, but it is not exactly the same code. Here we are logging the failure response from the backend. In the catch clause we are logging an exception thrown during the network call or the json parse. I will not make this change.